Documentation/security/threat-model.rst

```diff
@@ -0,0 +1,751 @@
+|                 |                     | connectivity will prevent an |
+|                 |                     | attacker from observing the  |
+|                 |                     | traffic intended for other   |
+|                 |                     | workloads, or sending        |
+|                 |                     | traffic that “spoofs” the    |
+|                 |                     | identity of another pod,     |
+|                 |                     | even if transparent          |
+|                 |                     | encryption is not in use.    |
+|                 |                     | Pods cannot send traffic     |
+|                 |                     | that “spoofs” other pods due |
+|                 |                     | to limits on the use of      |
+|                 |                     | source IPs and limits on     |
+|                 |                     | sending tunneled traffic.    |
++-----------------+---------------------+------------------------------+
+| Observability   | None                | Cilium’s Hubble flow-event   |
+| data            |                     | observability can be used to |
+|                 |                     | provide reliable audit of    |
+|                 |                     | the attacker’s L3/L4 and L7  |
+|                 |                     | network connectivity.        |
++-----------------+---------------------+------------------------------+
+
+.. _bandwidth limitations: https://docs.cilium.io/en/stable/network/kubernetes/bandwidth-manager/
+
+Recommended Controls
+^^^^^^^^^^^^^^^^^^^^
+
+-  Kubernetes workloads should have `defined resource limits`_.
+   This will help in ensuring that Cilium is not starved of resources due to a misbehaving deployment elsewhere in a cluster.
+-  Cilium can be given prioritized access to system resources either via
+   Kubernetes, cgroups, or other controls.
+-  Runtime security solutions such as `Tetragon`_ should be deployed to 
+   ensure that container compromises can be detected as they occur.
+
+.. _defined resource limits: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+.. _Tetragon: https://github.com/cilium/tetragon
+
+.. _limited-privilege-host-attacker:
+
+Limited-privilege Host Attacker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this scenario, the attacker is someone with the ability to run
+arbitrary code with direct access to the host PID or network namespace
+(or both), but without “root” privileges that would allow them to
+disable Cilium components or undermine the eBPF and other kernel state
+Cilium relies on.
+
+This level of access could exist for a variety of reasons, including:
+
+-  Pods or other containers running in the host PID or network
+   namespace, but not with “root” privileges. This includes
+   ``hostNetwork: true`` and ``hostPID: true`` containers.
+-  Non-“root” SSH or other console access to a Kubernetes worker node.
```

Comment:
I'm going back and forth a bit in my head about whether to consistently use the Kubernetes terminology in cases like below (which helpfully matches the diagrams), or switch to generic terminology that would help to reinforce that none of the underlying mechanisms here are Kubernetes-specific.

Example:

```suggestion
-  Non-“root” SSH or other console access to a node.
```

(I'm also fine with leaving this as-is, and if we want to improve this to be more generic in a subsequent set of changes we can do so after this PR)

Reply:
tgraf had a similar comment in the original document, so I think let's drop it from here.